edgexfoundry/device-mqtt-go
# feat(snap): support service secrets file config

This change would enable setting snap config arguments for the secrets file feature added by https://github.com/edgexfoundry/go-mod-bootstrap/pull/276.

It adds support for the following config items:
```
secret-store.secrets-file               // SecretStore.SecretsFile
secret-store.disable-scrub-secrets-file // SecretStore.DisableScrubSecretsFile
```

Package updates are added by https://github.com/canonical/edgex-snap-hooks/pull/13.

There are also some minor updates and fixes to the snap/README.md

Signed-off-by: Farshid Tavakolizadeh <farshid.tavakolizadeh@canonical.com>








## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have opened a PR for the related docs change - in the same PR

## Testing Instructions
1. build
```
snapcraft clean
snapcraft
```

2. Install, copy token, set conf, start, check logs:
```
$ sudo snap install ./edgex-device-mqtt_2.0.1-dev.11_amd64.snap --dangerous
$ sudo install -vDT /var/snap/edgexfoundry/current/secrets/device-mqtt/secrets-token.json /var/snap/edgex-device-mqtt/current/device-mqtt/secrets-token.json
$ sudo snap set edgex-device-mqtt env.secret-store.secrets-file=test.json
$ sudo snap start edgex-device-mqtt.device-mqtt
$ sudo snap logs edgex-device-mqtt.device-mqtt -n 100 | grep test.json
2021-10-28T09:49:57Z edgex-device-mqtt.device-mqtt[82733]: ++ export SECRETSTORE_SECRETSFILE=test.json
2021-10-28T09:49:57Z edgex-device-mqtt.device-mqtt[82733]: ++ SECRETSTORE_SECRETSFILE=test.json
2021-10-28T09:49:57Z edgex-device-mqtt.device-mqtt[82733]: level=INFO ts=2021-10-28T09:49:57.154987344Z app=device-mqtt source=variables.go:352 msg="Variables override of 'SecretStore.SecretsFile' by environment variable: SECRETSTORE_SECRETSFILE=test.json"
2021-10-28T09:49:57Z edgex-device-mqtt.device-mqtt[82733]: level=ERROR ts=2021-10-28T09:49:57.156765124Z app=device-mqtt source=bootstrap.go:46 msg="failed to create SecretProvider: seeding secrets failed: open test.json: no such file or directory"
```

## New Dependency Instructions (If applicable)
